File: code/backend/java/core/src/main/java/games/paths/core/model/story/CardInfo.java
package games.paths.core.model.story;

/**
 * CardInfo - Domain model for a visual card with resolved text
 * and optional creator information. Immutable record.
 *
 * <p>Used within story summaries and content detail APIs. Enhanced in Step 16
 * to include copyright and creator fields (consolidated from the former
 * CardDetail type).</p>
 */
public record CardInfo(
        String uuid,
        String cardType,
        String urlImage,
        String alternativeImage,
        String awesomeIcon,
        String styleMain,
        String styleDetail,
        String styleImageLittle,
        String styleImageMedium,
        String styleImageLarge,
        String title,
        String description,
        String copyrightText,
        String linkCopyright,
        CreatorInfo creator) {
}

File: code/backend/php/src/Core/Domain/Story/CardInfo.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Domain\Story;

class CardInfo
{
    public function __construct(
        public string $uuid,
        public ?string $cardType = null,
        public ?string $urlImage = null,
        public ?string $alternativeImage = null,
        public ?string $awesomeIcon = null,
        public ?string $styleMain = null,
        public ?string $styleDetail = null,
        public ?string $styleImageLittle = null,
        public ?string $styleImageMedium = null,
        public ?string $styleImageLarge = null,
        public ?string $title = null,
        public ?string $description = null,
        public ?string $copyrightText = null,
        public ?string $linkCopyright = null,
        public ?CreatorInfo $creator = null
    ) {
    }
}
